### DISCUSSION TITLE
Remove node if item is null

### DISCUSSION THREAD

**Discussion by andrepaulo-bit** [NaturalIntelligence/fast-xml-parser#434] [Q&A]
Hello, I was wondering if it's possible to remove a node if the value its null.

I had a look at suppressEmptyNode but it doesn't seem like it solves my issue.
Saw this issue: https://github.com/NaturalIntelligence/fast-xml-parser/issues/123
And that would work as well, but there are a few values that can be null, and I'm not sure how I can do that for every value.

**Comment by amitguptagwl** [comment]
Can you please explain with example?

**Reply by andrepaulo-bit** [reply]
Sure, for a webservice that I'm using, it accepts some fields as null, and whenever I send the node empty it claims there was a problem parsing the number.
So if I have this:
```json
{
    "requiredNumber": 4,
    "requiredString": "test",
    "optionalNumber": null,
}
```
It gets converted to this: 
`<requiredNumber>4</requiredNumber><requiredString>test</requiredString><optionalNumber/>`

But the webservice will complain that it can't parse optionalNumber.
If I add `xsi:nil="true"` to the field it will work, but if I just remove it from the XML it will also accept it.

`<requiredNumber>4</requiredNumber><requiredString>test</requiredString>`

So this would also be allowed on the webservice side.

I was hoping there was a easy way to filter out the empty tags to avoid going into each one and if its null add the `xsi:nil="true"`

**Reply by amitguptagwl** [reply]
hmm, got your point. So you're basically looking for a feature where you can decide when a tag should not be skipped.

**Reply by andrepaulo-bit** [reply]
yeah basically.
I tried doing what was in #123 but had a few problems, and considering that was almost 4 years ago went for a simpler option, with regex replace `<optionalNumber/>` and other optional parameters that I might have. Seems to do the trick for me.

**Comment by luiscarlosjayk** [comment]
I had same problem and what ended doing was running using RegExp to solve it:

```js
function removeEmptyNodesFromXML(xmlString) {
    const emptyNodesPattern = new RegExp(/(<[\w]+><[\w]+\/><\/[\w]+>)|(<[\w]+\/>)/gi);
    const cleanedXmlString = xmlString.replace(emptyNodesPattern, '');

    return cleanedXmlString;
}
```

This will remove any `<tag/>` with `<[\w]+\/>` and also it's parent if it's the only children node `<parent><node/></parent>` with `<[\w]+><[\w]+\/><\/[\w]+>`, they have a pipe so it makes an **or**.
That solved my case, however, if you need deeper/more nested traversal this won't work.
